src/utils/utils.ts

```diff
@@ -154,14 +154,16 @@ function promisify<T>(f): (...args: any[]) => Promise<T> {
   };
 }
 
-/**
- * Deconstructed promise
- */
-function promise<T>(): {
+export type PromiseType<T> = {
```

Comment:
1. The `export` should be at the very end.
2. This is a nominal type for the deconstructed promise, prefer calling this `PromiseDeconstructed`.